presto-main/src/main/java/com/facebook/presto/execution/QueryManagerConfig.java

```diff
@@ -159,6 +161,20 @@ public QueryManagerConfig setMaxQueryHistory(int maxQueryHistory)
         return this;
     }
 
+    @Min(1000)
```

Comment:
It seems arbitrary, however, and could preclude setting it smaller for testing. The purpose of constraints is to prevent out-of-range values that would break the system. In this case, `@Min(0)` is a legitimate value (with the expected outcome that every query will fail).